examples/example-jwt-auth/src/main/java/io/grpc/examples/jwtauth/AuthClient.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.jwtauth;
+
+import io.grpc.CallCredentials;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+/**
+ * An authenticating client that requests a greeting from the {@link AuthServer}.
+ */
+public class AuthClient {
+
+  private static final Logger logger = Logger.getLogger(AuthClient.class.getName());
+
+  private final ManagedChannel channel;
+  private final GreeterGrpc.GreeterBlockingStub blockingStub;
+
+  /**
+   * Construct client for accessing GreeterGrpc server.
+   */
+  AuthClient(String host, int port) {
+    this(ManagedChannelBuilder
+        .forAddress(host, port)
+        // Channels are secure by default (via SSL/TLS). For this example we disable TLS to avoid
+        // needing certificates, but it is recommended to use a secure channel while passing
+        // credentials.
+        .usePlaintext()
+        .build());
+  }
+
+  /**
+   * Construct client for accessing GreeterGrpc server using the existing channel.
+   */
+  AuthClient(ManagedChannel channel) {
+    this.channel = channel;
+    this.blockingStub = GreeterGrpc.newBlockingStub(channel);
+  }
+
+  public void shutdown() throws InterruptedException {
+    channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+  }
+
+  /**
+   * Say hello to server.
+   *
+   * @param name name to set in HelloRequest
+   * @param clientId client identifier to set in JWT subject
+   * @return the message in the HelloReply from the server
+   */
+  public String greet(String name, String clientId) {
+    logger.info("Will try to greet " + name + " ...");
+    HelloRequest request = HelloRequest.newBuilder().setName(name).build();
+
+    String jwt = getJwt(clientId); // build JWT
+    CallCredentials credentials = new BearerToken(jwt); // Wrap JWT in CallCredentials
```

Comment:
To make the example more powerful can we inject the bearer-token/call-credential in a client-interceptor? That might be a common or valuable use-case. The clientId itself comes from the client (say as a header) and the interceptor converts it into the bearer token.

Reply:
We could populate authorization header using `ClientInterceptor`, and then we do not even need `CallCredential` implementation, but the `withCallCredentials(CallCredentials credentials)` method looks more appropriate.

Reply:
Okay. The client-interceptor could be another example